api/controllers/link-controller.ts

```diff
@@ -29,7 +34,19 @@ export const createLink = async (
     })
     .toArray();
 
-  if (result.length !== 0) return createLink(longUrl, ipAddress);
+  if (result.length !== 0) {
+    if (!customCode) return createLink(longUrl, ipAddress);
```

Comment:
No need to use if and else with return. Return will stop further execution anyway.